README.md

```diff
@@ -51,13 +113,78 @@ logger.add_input_files({'input_csv': 'input.csv'})
 b1 = some_ground_breaking_research_result
 b2 = another_ground_breaking_research_result
 
-logger.add_output_files({'output_csv': 'output.csv'})
 logger.add_output_params({'b1': b1,
                           'b2': b2})
 ```
-**Note:** We don't support all variable types yet. When an unsupported type is passed as input/output param, a warning will be issued when the code is executed, and the param will be ignored.
+In this method - a dict with param name and value pairs is explicitly provided to the logger.
+
+**Pros**
+* Only params deemed relevant by you are captured
+* Params can be captured anywhere in the script (or other modules), as the logger object is passed and the proper functions are called
+
+**Cons**
+* You need to update the logging code as you update the script
+
+**Recommended use case:** When you need to log params that aren't in your global scope (in functions, modules, etc...).
+
+#### Combining Param Logging Methods
+It is possible to combine all the above mentioned methods  together to log params in your scripts.
+The manual and in/out blocks logging takes precedence over the automatic logging, since automatic logging doesn't log a param already known to the logger.
```

Comment:
I do not know what is the convention - but wouldn't a tired list of precedence be more appropriate here ?

Reply:
That's why we have an example. It's too complicated for words

Reply:
Cool...